packages/badge-service/badgeGenerator/badgeGeneratorSpec.ts

```diff
@@ -0,0 +1,193 @@
+import { assert, expect } from "chai";
+import * as sinon from 'sinon';
+import * as chai from "chai";
+import * as sinonchai from "sinon-chai";
+import * as dataAccessModule from "stryker-dashboard-data-access";
+import { Mock } from '../testHelpers/mock';
+import { MutationScoreMapper, MutationScore } from "stryker-dashboard-data-access";
+import { run } from "../badgeGenerator/badgeGenerator";
+import * as badgeGenerator from "../badgeGenerator/badgeGenerator";
+import * as httpHelpers from "../helpers/httpHelpers";
+
+chai.use(sinonchai);
+
+describe('Generating a badge', () => {
+
+    let sandbox: sinon.SinonSandbox;
+    let mutationScoreMapperMock: Mock<MutationScoreMapper>;
+    let context: any;
+    let req: any;
+    let mutationScore: MutationScore;
+
+    beforeEach(() => {
+        sandbox = sinon.createSandbox();
+        mutationScoreMapperMock = sinon.createStubInstance(MutationScoreMapper);
+        sandbox.stub(dataAccessModule, 'MutationScoreMapper').returns(mutationScoreMapperMock);
+
+        mutationScore = new MutationScore();
+        mutationScore.branch = 'master';
+        mutationScore.slug = 'github/stryker-mutator/stryker';
+        mutationScore.score = 97.8;
+
+        let mutationScoreNoBranch = new MutationScore();
+        mutationScoreNoBranch.slug = 'github/stryker-mutator/stryker';
+        mutationScoreNoBranch.score = 79;
+
+        mutationScoreMapperMock.selectSingleEntity.withArgs('github/stryker-mutator', 'stryker/master')
+            .resolves(mutationScore);
+        mutationScoreMapperMock.selectSingleEntity.withArgs('github/stryker-mutator', 'stryker')
+            .resolves(mutationScoreNoBranch);
+
+        context = {
+            log: () => { },
+            bindingData: {
+                provider: 'github',
+                owner: 'stryker-mutator',
+                repo: 'stryker',
+                branch: 'master'
+            }
+        };
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+    });
+
+    it('Should return a http 400 error when slug can not be found', async () => {
+        // Arrange
+        context.bindingData.owner = 'owner-does-not-exist';
+
+        // Act
+        await run(context, req);
+
+        // Assert
+        expect(context.res.status).to.equal(400);
+    }),
+        it('Should return a http 400 error when the combination of slug & branch can not be found', async () => {
+            // Arrange
+            context.bindingData.branch = 'dev';
+
+            // Act
+            await run(context, req);
+
+            // Assert
+            expect(context.res.status).to.equal(400);
+        }),
+        it('Should return a http 500 error when storage can not be reached', async () => {
```

Comment:
On error status codes we might want to resort to https://img.shields.io/badge/mutation%20score-unknown-lightgrey.svg, so the result is still a badge in your readme.

Reply:
Same as above, I'll make that change.

Reply:
👍